src/Database/LSMTree/Internal/Merge.hs

```diff
@@ -113,8 +131,12 @@ complete Merge{..} = do
     readMutVar mergeLastStepResult >>= \case
       MergeInProgress -> error "complete: Merge is not yet completed!"
       MergeComplete -> do
+        -- Since access to a merge /should/ be sequentialised, we can assume
+        -- that the ref count has not changed between this read and the use of
+        -- fromMutable.
+        n <- RC.readRefCount mergeRefCounter
```

Comment:
Is it definately sequentialised? What about duplicating or closing other table handles that share this merging run?

Do we need to copy the ref count value? How about reusing/sharing the mutable refcount itself? Then it's safe wrt other threads modifying the count.

Reply:
Addressed in #423